examples/hello/bpReader/helloBPReader.cpp

```diff
@@ -64,12 +74,15 @@ int main(int argc, char *argv[])
             bpIO.InquireVariable<float>("bpFloats");
         adios2::Variable<int> bpInts = bpIO.InquireVariable<int>("bpInts");
 
+        unsigned long  Nx = 10;
         if (bpFloats) // means found
         {
+            std::vector<float> myFloats;
+
             // read only the chunk corresponding to our rank
             bpFloats.SetSelection({{Nx * rank}, {Nx}});
             // myFloats.data is pre-allocated
```

Comment:
Remove the comment since `std::vector` is not pre-allocated anymore

Reply:
So, I like this this, moving the vector to a more local scope.  However, since the example has two variables / vectors that reads it would be useful since the API allows it to use both allocation methods rather than change them both to be post-allocated like this.  So when moving to a more local scope, can you restore one of them to still be preallocated?

Reply:
@chuckatkins thanks for bringing that up. Since Gets are in sync mode, this is fine. For Deferred mode is not going to work as the vectors go out of scope at `EndStep()`. This is good example that illustrates the Put memory contracts.

Reply:
@williamfgc : Nice catch, done.